File: examples/hello_world/hello_world.py
from __future__ import print_function
import os,sys
import mlflow

print("MLflow Version:", mlflow.version.VERSION)
print("Tracking URI:", mlflow.tracking.get_tracking_uri())
experiment_name = os.environ.get("MLFLOW_EXPERIMENT_NAME","hello_world")
print("experiment_name:",experiment_name)

def run(alpha, run_origin, log_artifact):
    mlflow.set_experiment(experiment_name)
    with mlflow.start_run(run_name=run_origin) as run:  # NOTE: mlflow CLI ignores run_name
        print("runId:",run.info.run_uuid)
        print("artifact_uri:",mlflow.get_artifact_uri())
        print("alpha:",alpha)
        print("log_artifact:",log_artifact)
        print("run_origin:",run_origin)
        mlflow.log_param("alpha", alpha)
        mlflow.log_metric("rmse", 0.786)
        mlflow.set_tag("mlflow.runName",run_origin) # mlflow CLI picks this up
        mlflow.set_tag("run_origin", run_origin)
        mlflow.set_tag("log_artifact", log_artifact)
        if log_artifact:
            with open("info.txt", "w") as f:
                f.write("Hi artifact")
            mlflow.log_artifact("info.txt")

if __name__ == "__main__":
    alpha = sys.argv[1] if len(sys.argv) > 1 else "0.1"
    run_origin = sys.argv[2] if len(sys.argv) > 2 else "None"
    log_artifact = bool(sys.argv[3]) if len(sys.argv) > 3 else False
    run(alpha,run_origin,log_artifact)

File: examples/scala/src/main/scala/org/andre/mlflow/examples/hello/HelloWorld.scala
package org.andre.mlflow.examples.hello

import java.io.{File,PrintWriter}
import org.mlflow.tracking.MlflowClient
import org.mlflow.api.proto.Service.{RunStatus,CreateRun}
import scala.collection.JavaConversions._
import org.andre.mlflow.util.MLflowUtils

object HelloWorld {
  def main(args: Array[String]) {

    // Create MLflow client
    val client = MLflowUtils.createMlflowClient(args)

    // Create or get existing experiment
    val expName = "scala_HelloWorld"
    val expId = MLflowUtils.getOrCreateExperimentId(client, expName)
    println("Experiment name: "+expName)
    println("Experiment ID: "+expId)

    // Create run
    val runInfo = client.createRun(expId)
    val runId = runInfo.getRunUuid()
    println("Run ID: "+runId)

    // Log params and metrics
    client.logParam(runId, "alpha","0.5")
    client.logMetric(runId, "rmse",0.876)
    client.setTag(runId, "origin","laptop")
    client.setTag(runId, "mlflow.source.name",MLflowUtils.getSourceName(getClass())) // populates "Source" field in UI
    client.setTag(runId, "mlflow.runName","myRun") // populates "Run Name" field in UI

    // Log file artifact
    new PrintWriter("info.txt") { write("File artifact: "+new java.util.Date()) ; close }
    client.logArtifact(runId, new File("info.txt"))

    // Log directory artifact
    val dir = new File("tmp")
    dir.mkdir
    new PrintWriter(new File(dir, "model.txt")) { write("Directory artifact: "+new java.util.Date()) ; close }
    client.logArtifacts(runId, dir, "model")

    // Close run
    client.setTerminated(runId, RunStatus.FINISHED, System.currentTimeMillis())
  }
}
